hash_reader.go

```diff
@@ -133,43 +172,98 @@ func (hashReader *hashReader) TimeValue() (time.Time, error) {
 }
 
 func (hashReader *hashReader) StringValue() (string, error) {
-	panic("implement me")
 	return hashReader.ionReader.StringValue()
 }
 
 func (hashReader *hashReader) ByteValue() ([]byte, error) {
-	panic("implement me")
 	return hashReader.ionReader.ByteValue()
 }
 
 func (hashReader *hashReader) Sum(b []byte) []byte {
-	panic("implement me")
+	hashReader.hasher.digest()
+}
+
+func (hashReader *hashReader) traverse() error {
+	for hashReader.Next() {
+		switch hashReader.typeIon {
```

Comment:
why are we using switch statement when you are only performing some action on one type

Reply:
We perform action if that type is one of the container types (ie. list, sexp, struct). The reasoning is the same as the switch statement in hashReader.Next()